### ISSUE TITLE
Trimmomatic log files are ignored

### ISSUE THREAD

**Issue by patrickvdb** [ewels/MultiQC#539]
**Description of bug:**
Some of my trimmomatic log files are ignored by multiqc. It seems like the file sizes are too large, because in the cases where it does work the files are the smallest (see below)

**Error log:**
Log files will say that all 12 log files (`trimlog.txt`) are ignored, but also that it finds 5 log files.

```
$ cat multiqc_data/multiqc.log 
[2017-08-02 13:49:36,791] multiqc                                            [DEBUG  ]  No MultiQC config found: /opt/conda/lib/python2.7/site-packages/multiqc-1.0.dev0-py2.7.egg/multiqc_config.yaml
[2017-08-02 13:49:36,792] multiqc                                            [DEBUG  ]  No MultiQC config found: /home/sokolova/.multiqc_config.yaml
[2017-08-02 13:49:36,792] multiqc                                            [DEBUG  ]  No MultiQC config found: multiqc_config.yaml
[2017-08-02 13:49:36,792] multiqc                                            [DEBUG  ]  Command used: /opt/conda/bin/multiqc -m trimmomatic -m fastqc -m star -f .
[2017-08-02 13:49:36,792] multiqc                                            [DEBUG  ]  Could not connect to multiqc.info for version check
[2017-08-02 13:49:36,792] multiqc                                            [INFO   ]  This is MultiQC v1.0.dev0
[2017-08-02 13:49:36,792] multiqc                                            [DEBUG  ]  Command     : /opt/conda/bin/multiqc -m trimmomatic -m fastqc -m star -f .
[2017-08-02 13:49:36,792] multiqc                                            [DEBUG  ]  Working dir : /home/sokolova/patrick/S2TM2T_0717
[2017-08-02 13:49:36,792] multiqc                                            [INFO   ]  Template    : default
[2017-08-02 13:49:36,792] multiqc                                            [DEBUG  ]  Running in --force mode, will overwrite any existing reports.
[2017-08-02 13:49:36,792] multiqc                                            [INFO   ]  Searching '.'
[2017-08-02 13:49:36,792] multiqc                                            [INFO   ]  Only using modules trimmomatic, fastqc, star
[2017-08-02 13:49:36,793] multiqc                                            [DEBUG  ]  Analysing modules: star, trimmomatic, fastqc
[2017-08-02 13:49:36,793] multiqc                                            [DEBUG  ]  Using temporary directory for creating report: /tmp/tmpHhq68x
[2017-08-02 13:49:36,796] multiqc                                            [DEBUG  ]  Ignoring directory as matched fn_ignore_dirs: ./multiqc_data
[2017-08-02 13:49:36,936] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-001_TTACCGAC.fq.gz
[2017-08-02 13:49:36,937] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-002_TCGTCTGA.fq.gz
[2017-08-02 13:49:36,937] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-003_TTCCAGGT.fq.gz
[2017-08-02 13:49:36,937] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-004_TACGGTCT.fq.gz
[2017-08-02 13:49:36,937] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-005_AAGACCGT.fq.gz
[2017-08-02 13:49:36,937] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-006_CAGGTTCA.fq.gz
[2017-08-02 13:49:36,937] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-007_TAGGAGCT.fq.gz
[2017-08-02 13:49:36,937] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-008_TACTCCAG.fq.gz
[2017-08-02 13:49:36,937] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-009_AGTGACCT.fq.gz
[2017-08-02 13:49:36,937] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-010_AGCCTATC.fq.gz
[2017-08-02 13:49:36,937] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-011_TCATCTCC.fq.gz
[2017-08-02 13:49:36,937] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-012_CCAGTATC.fq.gz
[2017-08-02 13:49:36,951] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-008_TACTCCAG.fq.gz
[2017-08-02 13:49:36,951] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,952] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-009_AGTGACCT.fq.gz
[2017-08-02 13:49:36,952] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,953] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-010_AGCCTATC.fq.gz
[2017-08-02 13:49:36,953] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,954] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-004_TACGGTCT.fq.gz
[2017-08-02 13:49:36,954] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,954] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-003_TTCCAGGT.fq.gz
[2017-08-02 13:49:36,954] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,955] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-006_CAGGTTCA.fq.gz
[2017-08-02 13:49:36,955] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,956] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-007_TAGGAGCT.fq.gz
[2017-08-02 13:49:36,956] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,956] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-001_TTACCGAC.fq.gz
[2017-08-02 13:49:36,957] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,957] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-002_TCGTCTGA.fq.gz
[2017-08-02 13:49:36,957] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,958] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-011_TCATCTCC.fq.gz
[2017-08-02 13:49:36,958] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,959] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-005_AAGACCGT.fq.gz
[2017-08-02 13:49:36,959] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,960] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-012_CCAGTATC.fq.gz
[2017-08-02 13:49:36,960] multiqc                                            [DEBUG  ]  Ignoring file as too large: trimlog.txt
[2017-08-02 13:49:36,963] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-008_TACTCCAG.genome.bam
[2017-08-02 13:49:36,963] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-008_TACTCCAG.transcript.bam
[2017-08-02 13:49:36,965] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-009_AGTGACCT.genome.bam
[2017-08-02 13:49:36,965] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-009_AGTGACCT.transcript.bam
[2017-08-02 13:49:36,967] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-003_TTCCAGGT.genome.bam
[2017-08-02 13:49:36,967] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-003_TTCCAGGT.transcript.bam
[2017-08-02 13:49:36,969] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-001_TTACCGAC.genome.bam
[2017-08-02 13:49:36,969] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-001_TTACCGAC.transcript.bam
[2017-08-02 13:49:36,971] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-006_CAGGTTCA.genome.bam
[2017-08-02 13:49:36,971] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-006_CAGGTTCA.transcript.bam
[2017-08-02 13:49:36,973] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-011_TCATCTCC.genome.bam
[2017-08-02 13:49:36,973] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-011_TCATCTCC.transcript.bam
[2017-08-02 13:49:36,975] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-004_TACGGTCT.genome.bam
[2017-08-02 13:49:36,975] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-004_TACGGTCT.transcript.bam
[2017-08-02 13:49:36,977] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-002_TCGTCTGA.genome.bam
[2017-08-02 13:49:36,977] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-002_TCGTCTGA.transcript.bam
[2017-08-02 13:49:36,979] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-010_AGCCTATC.genome.bam
[2017-08-02 13:49:36,979] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-010_AGCCTATC.transcript.bam
[2017-08-02 13:49:36,981] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-007_TAGGAGCT.genome.bam
[2017-08-02 13:49:36,981] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-007_TAGGAGCT.transcript.bam
[2017-08-02 13:49:36,983] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-005_AAGACCGT.genome.bam
[2017-08-02 13:49:36,983] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-005_AAGACCGT.transcript.bam
[2017-08-02 13:49:36,985] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-012_CCAGTATC.genome.bam
[2017-08-02 13:49:36,985] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: 103131-001-012_CCAGTATC.transcript.bam
[2017-08-02 13:49:36,996] multiqc                                            [DEBUG  ]  Ignoring directory as matched fn_ignore_dirs: ./multiqc/multiqc_data
[2017-08-02 13:49:37,467] multiqc.modules.star.star                          [DEBUG  ]  Could not find any reports in ('.',)
[2017-08-02 13:49:41,427] multiqc.modules.trimmomatic.trimmomatic            [INFO   ]  Found 5 logs
[2017-08-02 13:49:45,436] multiqc.modules.fastqc.fastqc                      [INFO   ]  Found 12 reports
[2017-08-02 13:49:46,554] multiqc                                            [WARNING]  Deleting    : multiqc_report.html   (-f was specified)
[2017-08-02 13:49:46,562] multiqc                                            [INFO   ]  Report      : multiqc_report.html
[2017-08-02 13:49:46,562] multiqc                                            [WARNING]  Deleting    : multiqc_data   (-f was specified)
[2017-08-02 13:49:46,567] multiqc                                            [INFO   ]  Data        : multiqc_data
[2017-08-02 13:49:46,567] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpHhq68x/multiqc_data/multiqc_fastqc.txt' to '/home/sokolova/patrick/S2TM2T_0717/multiqc_data'
[2017-08-02 13:49:46,571] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpHhq68x/multiqc_data/multiqc_trimmomatic.txt' to '/home/sokolova/patrick/S2TM2T_0717/multiqc_data'
[2017-08-02 13:49:46,573] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpHhq68x/multiqc_data/multiqc_general_stats.txt' to '/home/sokolova/patrick/S2TM2T_0717/multiqc_data'
[2017-08-02 13:49:46,575] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpHhq68x/multiqc_data/multiqc_sources.txt' to '/home/sokolova/patrick/S2TM2T_0717/multiqc_data'
[2017-08-02 13:49:46,713] multiqc                                            [INFO   ]  MultiQC complete
```
**More file info:**
The files that it ends up using are the 5 smallest (below about 1.1GB)

File sizes:
```
$ du -hs `ls fq_trim/*/trimlog.txt`
1.6G    fq_trim/103131-001-001_TTACCGAC/trimlog.txt
1.1G    fq_trim/103131-001-002_TCGTCTGA/trimlog.txt
1.4G    fq_trim/103131-001-003_TTCCAGGT/trimlog.txt
1.8G    fq_trim/103131-001-004_TACGGTCT/trimlog.txt
1020M   fq_trim/103131-001-005_AAGACCGT/trimlog.txt
1.3G    fq_trim/103131-001-006_CAGGTTCA/trimlog.txt
1.1G    fq_trim/103131-001-007_TAGGAGCT/trimlog.txt
1.1G    fq_trim/103131-001-008_TACTCCAG/trimlog.txt
1.4G    fq_trim/103131-001-009_AGTGACCT/trimlog.txt
1.9G    fq_trim/103131-001-010_AGCCTATC/trimlog.txt
1.5G    fq_trim/103131-001-011_TCATCTCC/trimlog.txt
982M    fq_trim/103131-001-012_CCAGTATC/trimlog.txt
```

Used samples:
```
$ cat multiqc_data/multiqc_trimmomatic.txt 
Sample  input_reads     surviving_pct   dropped surviving       dropped_pct
103131-001-002_TCGTCTGA 16750754.0      99.12   147756.0        16602998.0      0.88
103131-001-003_TTCCAGGT 21316142.0      99.13   185229.0        21130913.0      0.87
103131-001-005_AAGACCGT 15903659.0      99.15   134997.0        15768662.0      0.85
103131-001-007_TAGGAGCT 16183074.0      99.12   142423.0        16040651.0      0.88
103131-001-012_CCAGTATC 15321164.0      99.1    137393.0        15183771.0      0.9
```

* **Method of MultiQC installation:**
Singularity image: https://singularity-hub.org/collections/310/
which is based on this docker image: https://hub.docker.com/r/pvdb90/multiqc/

**Comment by ewels** [comment]
Wow, those are big log files! :) Thankfully, you can override the default threshold at which MultiQC deems a file too big to parse. Just change the `log_filesize_limit` config option. For example, in `~/.multiqc_config.yaml`:

```yaml
log_filesize_limit: 2000000000
```

I was going to point you to the part of the documentation saying this, but it seems as though it has slipped the net and isn't described anywhere. Sorry about that! I'll add it now.

Phil

**Comment by ewels** [comment]
Now [documented](http://multiqc.info/docs/#big-log-files) - thanks again!

**Comment by patrickvdb** [comment]
Thank you as well!

**Comment by lczech** [comment]
Same issue here, a lot of files are ignored: 
[multiqc.log](https://github.com/ewels/MultiQC/files/5121334/multiqc.log)

However, none of them is larger than 600kb. Why is that happening? Also, a lot of `qualimap` files are apparently ignored due to file size, but they don't seem relevant at all, such as 

    File ignored by longranger/invocation because it exceeded search pattern filesize limit: underscore.js

Is that just a non-fitting log message, or is something else going on here?

**Comment by ewels** [comment]
Hi @lczech,

MultiQC skips all large files as an initial step before it checks them against the module search patterns, as a filesize check is very very fast but some of the module searches are slow (eg. checking each file contents for a pattern).

Each module can specify a maximum file size (see [docs](https://multiqc.info/docs/#step-1-find-log-files)) so each file can be checked multiple times (with different thresholds). In your pasted log above, the search pattern in question is `longranger/invocation`, which you can see documented [here](https://multiqc.info/docs/#longranger):

```yaml
longranger/invocation:
  fn: _invocation
  contents: call PHASER_SVCALLER_CS(
  max_filesize: 2048
```

So this particular search pattern is skipping any files which are > 2048 bytes.

Basically, this log message can nearly always be ignored. It's only useful when you have a specific file being ignored by a specific module when it shouldn't be. Then that log file should be able to tell you why.

Phil

**Comment by lczech** [comment]
Thanks for the clarification! So, increasing the max file size would solve my issue then? 
(I would try on my own, but have moved on with the project in the meantime, so cannot easily reproduce as of now - but for future reference and others who might find this thread, would be good to know how to include those files ;-))

**Comment by ewels** [comment]
I'm not totally clear on what your issue was - the log message is as expected, MultiQC is working properly - right? Did MultiQC ever fail to report something you expected? Basically everything in your [original comment](https://github.com/ewels/MultiQC/issues/539#issuecomment-679600959) is as it should be, nothing is wrong.

**Comment by lczech** [comment]
Hm, I don't exactly remember, and as said, unfortunately, cannot reproduce any more. But I think my issue was that files were ignored due to file size limits (according to those log messages), although those files are smaller than the limit. So that is either a bug, or a misleading message about why they are ignored.

But I think we can ignore this for now. If I ever run into this again, I'll let you know! 

Thanks for the support!
